activerecord/lib/active_record/connection_adapters.rb

```diff
@@ -4,6 +4,63 @@ module ActiveRecord
   module ConnectionAdapters
     extend ActiveSupport::Autoload
 
+    @adapters = {}
+
+    class << self
+      # Registers a custom database adapter.
+      #
+      # Can also be used to define aliases.
+      #
+      # == Example
+      #
+      #   ActiveRecord::ConnectionAdapters.register("megadb", "MegaDB::ActiveRecordAdapter", "mega_db/active_record_adapter")
+      #
+      #   ActiveRecord::ConnectionAdapters.register("mysql", "ActiveRecord::ConnectionAdapters::TrilogyAdapter", "active_record/connection_adapters/trilogy_adapter")
+      #
+      def register(name, class_name, path)
+        @adapters[name] = [class_name, path]
```

Comment:
WDYT about a class here? 

```ruby
@adapters[name] = RegisteredAdapter.new(name:, class_name:, path:)
```

Gives it a wrapper class to add methods to validate if `path` comes in as a string vs a `Pathname`, or even allowing `class_name` to be a proc with a class in it, etc..

Only a suggestion. Since it's local to this module and not exposed it's not a big deal to manage an array with positional args it's basically an implementation detail.

Reply:
I don't think it's really worth it, the logic is simple enough that a Tuple is fine. If for some reason this becomes more complicated, then yes, why not a class.